pkg/lib/infra/task/task.go

```diff
@@ -4,19 +4,23 @@ import (
 	"context"
 )
 
+type Param interface {
+	TaskName() string
+}
+
 type Spec struct {
 	Name  string
-	Param interface{}
+	Param Param
 }
```

Comment:
The struct `Spec` now seems obsolete. We can just pass the `Param` interface around.